### PR TITLE
Integrates shared library go-commons - ES indexing

### LINKED ISSUE #115: Integrate open search use case with go-commons
Request to integrate or move all the opensearch api related logic used for ES indexing to https://github.com/cloud-bulldozer/go-commons

### PR DESCRIPTION
**Pull request by chentex**
### Description

Removes local implementation of ES indexer and favors `go-commons` library.

### Fixes

Fix #115

### CODE REVIEW

**Comment by vishnuchalla** [comment]
Thanks for this integration. From [here](https://github.com/cloud-bulldozer/ocm-api-load/blob/ab6c26ba3076cdad05f70b85da8886514e69f89c/pkg/elastic/elastic.go#LL37C68-L37C76), It looks like we have results in a file and then we are getting it in memory and indexing multiple docs to ES. Instead can't we simply have all the data in memory and pass that list to the indexer to index all at once? just trying to see if we can get rid of that file if it is redundant. And also can you paste the content of the file here just for a rought idea, please!

**Comment by rsevilla87** [comment]
> Thanks for this integration. From [here](https://github.com/cloud-bulldozer/ocm-api-load/blob/ab6c26ba3076cdad05f70b85da8886514e69f89c/pkg/elastic/elastic.go#LL37C68-L37C76), It looks like we have results in a file and then we are getting it in memory and indexing multiple docs to ES. Instead can't we simply have all the data in memory and pass that list to the indexer to index all at once? just trying to see if we can get rid of that file if it is redundant. And also can you paste the content of the file here just for a rought idea, please!

Apart from this, the library also includes a local file indexer in case you're interested on dumping the metrics in files.
https://github.com/cloud-bulldozer/go-commons/blob/main/indexers/local.go

**Comment by chentex** [comment]
As for the moment the file is not redundant, since is the file `vegeta` library pours all it's data to, up to 100,000 items in case of some tests scenarios, after that we load to file to apply some data to identify it and then push it to ES.

I can try having only a writer and see if it will hold everything in memory and the app doesn't crash.

File: pkg/elastic/elastic.go [hunk 1]
```diff
@@ -1,145 +1,70 @@
...
-		})
+	logger.Info(ctx, "Writing results to: %s", file)
+	data, err := json.Marshal(_doc)
+	if err != nil {
+		logger.Error(ctx, "Error during json marshal: %s", err)
+	} else {
+		err = os.WriteFile(file, data, 0666)
 		if err != nil {
-			errors = fmt.Sprintf("%s\n%s", errors, err)
+			logger.Error(ctx, "Error during ES indexing: %s", err)
```
Comment (vishnuchalla) [inline pkg/elastic/elastic.go:63]: Requesting correction on this log message.

File: pkg/tests/run.go [hunk 2]
```diff
@@ -142,26 +142,16 @@ func (r *Runner) Run(ctx context.Context) error {
...
-				r.logger.Info(ctx, "Results written to: %s", fileName)
-				err = resultsFile.Close()
+				// Index result file
+
+				serverVersion := helpers.GetServerVersion(ctx, conn)
+				r.logger.Info(ctx, "server version %s", serverVersion)
+				err = elastic.IndexFile(ctx, r.testID, serverVersion, testOptions.TestName, fileName, *testOptions.Metrics, r.logger)
 				if err != nil {
-					return err
+					r.logger.Error(ctx, "Error during ES indexing: %s", err)
```
Comment (vishnuchalla) [inline pkg/tests/run.go:151]: Same as the above.

**Review by vishnuchalla — APPROVED** [review]
lgtm

**Review by rsevilla87 — APPROVED** [review]
lgtm!

**Pull request merged**